src/Symfony/Component/Security/Core/Exception/SafeMessageAuthenticationException.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Exception;
+
+/**
+ * An authentication exception where you can control the message shown to the user.
+ *
+ * Be sure that the message passed to this exception is something that
+ * can be shown safely to your user. In other words, avoid catching
+ * other exceptions and passing their message directly to this class.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class SafeMessageAuthenticationException extends AuthenticationException
```

Comment:
I'm really not comfortable with the naming. The idea is that the message is safe to be displayed to the end user. So, what about `UserMessageAuthenticationException`, or `UserFacingAuthenticationException`, `UserOrientedAuthenticationException`? You get my point; the naming should indicate that the message is "**for end users**". My proposals are probably not the right ones, but we need to find a better name.